### PR TITLE
add onlyEncodeTransactions support

### PR DESCRIPTION
**Pull request by LuttyYang**
## Summary by CodeRabbit

- **New Features**
	- Introduced new methods for encoding transaction data for various smart contract functions, enhancing off-chain transaction preparation.
	- Added optional properties across multiple response types to include encoded transaction data and make transaction hashes optional.

- **Bug Fixes**
	- Improved transaction handling in dispute and permission management by introducing clearer request structures and error handling.

- **Documentation**
	- Updated type definitions to enhance clarity and flexibility in transaction responses, including optional fields for transaction hashes and encoded data.

- **Tests**
	- Added comprehensive test cases for the `ipAssetClient` functionality to ensure accurate handling of encoded transaction data during IP asset registration.

### CODE REVIEW

File: packages/core-sdk/src/types/resources/dispute.ts [hunk 1]
```diff
@@ -12,7 +13,8 @@ export type RaiseDisputeRequest = {
 };
 
 export type RaiseDisputeResponse = {
-  txHash: string;
+  txHash?: string;
+  encodedTx?: MethodEncode;
```
[the remark this answers is not in the document]
Comment (edisonz0718) [reply packages/core-sdk/src/types/resources/dispute.ts:17]: Can we rename this to `EncodedTxData`?

[the remark this answers is not in the document]
Comment (edisonz0718) [reply packages/core-sdk/src/types/resources/dispute.ts:17]: Also we need to add this to the `client.ts` so that users can import from their side

**Comment by bpolania** [comment]
Hello @jhd2best @LuttyYang 

* As mentioned before the right flow is to request a PR from the `[dev](https://github.com/storyprotocol/sdk/tree/dev)` branch and the it will be merged to `main`

* [This](https://github.com/storyprotocol/sdk/blob/main/package.json) `package.json` file that was updated is the wrong one, this is the top level file. The file that needs to be updated is the core-sdk file, you can find it [here](https://github.com/storyprotocol/sdk/blob/main/packages/core-sdk/package.json) 

Note that the current version is `1.0.0-rc.15` so it will need to be updated to `1.0.0-rc.16`, you can use as a reference the current version of the `npm` package [here](https://www.npmjs.com/package/@story-protocol/core-sdk)

**Review by bpolania — APPROVED** [review]

**Comment by bpolania** [comment]
I approved it, but please keep in mind my previous comments for the next PR

**Pull request merged**